### DISCUSSION TITLE
Flag for changing convention for arranging `m`

### DISCUSSION THREAD

**Discussion by mitkotak** [e3nn/e3nn-jax#82] [Ideas]
@curiosity54 and I were talking about how it would be good to have a flag that can let us go between different ordering conventions similar to how [e3x](https://e3x.readthedocs.io/stable/pitfalls.html) does it.

Making a list of all the different places where this flag would have to be propagated

- `e3nn.clebsch_gordon`
- `e3nn.spherical_harmonics`

This list will likely grow

**Comment by mariogeiger** [comment]
We already use a convention in which l=1 corresponds to a x,y,z (pseudo)vector.

**Reply by curiosity54** [reply]
Hey Mario! First I just wanted to confirm if the spherical harmonics defined here were with the Condon Shortley convention. Then we were wondering if we should also support the "regular" ordering of spherical harmonics, depending on the use case. I was expecting the Clebsch Gordan matrix would just be needed to account for the change of order between the two conventions. Would you say that is reasonable? Just wanted to double check

**Reply by curiosity54** [reply]
Another thing, which may just be a question because I probably rushed and overlooked some part of the documentation. From  e3nn.o3 import spherical_harmonics if I pass **x** the x,y and z coordinates in the columns, given the convention, for l=1 the  output keeps the order intact, x y z. So the convention in e3nn is indeed the one you describe. I saw a note that maybe I didnt fully understand about passing `normalization='integral'` , is that supposed to return the spherical harmonics in the "regular ordering (y z x)" or just normalize the output (which is what it seems to be doing)?

**Reply by ameya98** [reply]
Just normalizes the output so that the integral over S2 is 1!
